arkouda/pdarraycreation.py

```diff
@@ -196,13 +207,23 @@ def array(
 
     # If a is already a pdarray, do nothing
     if isinstance(a, pdarray):
+        # TODO update if max_bits is set
         return a if dtype is None else akcast(a, dtype)
     from arkouda.client import maxTransferBytes
 
     # If a is not already a numpy.ndarray, convert it
     if not isinstance(a, np.ndarray):
         try:
-            a = np.array(a, dtype=dtype)
+            if dtype == bigint:
+                tmp = np.array(a)
+                # numpy converts to float for large int (i.e. 2**64 - 1)
+                # avoid this so we don't lose precision
+                if tmp.dtype != np.float_:
+                    a = tmp
+                else:
+                    a = np.array(a, dtype=np.uint)
+            else:
+                a = np.array(a, dtype=dtype)
```

Comment:
we should hold off on merging this until after #1984, if people agree with that approach since this is a workaround for the same issue. Once that is merged this section should just be
```python
if dtype is not None and dtype != bigint:
```

Reply:
I updated this block to account for 1984, it should be good once CI clears